mcstatus/motd.py

```diff
@@ -0,0 +1,558 @@
+            See `this answer <https://stackoverflow.com/a/30313558>`_ as example.
+
+        :returns: String that contain HTML.
+        """
+        result = "<p>"
+        on_reset = ""
+        for element in self.parsed:
+            if isinstance(element, str):
+                result += element
+            elif isinstance(element, TranslationTag):
+                continue
+            elif isinstance(element, WebColor):
+                result += f"<span style='color:{element.hex}'>"
+                on_reset += "</span>"
+
+            elif isinstance(element, Formatting):
+                if element == Formatting.RESET:
+                    result += on_reset
+                    on_reset = ""
+                elif element == Formatting.OBFUSCATED:
+                    result += "<span class=obfuscated>"
+                    on_reset += "</span>"
+                else:
+                    tag_name = FORMATTING_TO_HTML_TAGS[element]
+                    result += "<" + tag_name + ">"
+                    on_reset += "</" + tag_name + ">"
+            else:
+                mc_color_to_hex = MINECRAFT_COLOR_TO_HEX_BEDROCK if self.bedrock else MINECRAFT_COLOR_TO_HEX_JAVA
+                foreground_color, background_color = mc_color_to_hex[element]
+
+                result += f"<span style='color:{foreground_color};text-shadow:0 0 1px {background_color}'>"
+                on_reset += "</span>"
+
+        return result + on_reset + "</p>"
+
+    @property
+    def xterm_256(self) -> str:
+        """Getter for a xterm-256 variant of MOTD.
+
+        The only difference from ``ansi`` is that it's using RGB colors instead of ``red``, ``blue`` etc.
+
+        :returns: String full of xterm-256 colors.
+        """
+        result = r"\e[0m"
+        for element in self.parsed:
+            if isinstance(element, str):
+                result += element
+            elif isinstance(element, TranslationTag):
+                continue
+            elif isinstance(element, WebColor):
+                result += r"\x1b[38;2;{0};{1};{2}m".format(*element.rgb)
+
+            elif isinstance(element, Formatting):
+                if element == Formatting.RESET:
+                    result += r"\e[0m"
+                else:
+                    tag_number = FORMATTING_TO_ANSI_TAGS[element]
+                    result += r"\e[" + tag_number + "m"
+            else:
+                mc_color_to_hex = MINECRAFT_COLOR_TO_HEX_BEDROCK if self.bedrock else MINECRAFT_COLOR_TO_HEX_JAVA
+                foreground_color, background_color = mc_color_to_hex[element]
+
+                result += r"\x1b[38;2;{0};{1};{2}m".format(
+                    *tuple(int(foreground_color.lstrip("#")[i : i + 2], 16) for i in (0, 2, 4))
+                )
+                result += r"\x1b[48;2;{0};{1};{2}m".format(
+                    *tuple(int(background_color.lstrip("#")[i : i + 2], 16) for i in (0, 2, 4))
+                )
```

Comment:
We can avoid the repetition of hex -> rgb logic by using `WebColor` to help us out here
```suggestion
                fg_hex, bg_hex  = mc_color_to_hex[element]
                fg_rgb = WebColor.from_hex(fg_hex).rgb
                bg_rgb = WebColor.from_hex(bg_hex).rgb

                result += r"\x1b[38;2;{0};{1};{2}m".format(*fg_rgb)
                result += r"\x1b[48;2;{0};{1};{2}m".format(*bg_rgb)
```

Reply:
Fixed [here](https://github.com/py-mine/mcstatus/compare/26690ce403010ab0af019cf2230d246e2920602e..ec2cacf3a1ee18ca0432a5c5e54c80ee6dbc3fcb), thanks!